source/numerics.tex

```diff
@@ -10397,3 +10398,5060 @@
+\end{codeblock}
+\end{itemdescr}
+
+\begin{itemdecl}
+template<class ExecutionPolicy, @\exposconcept{in-matrix}@ InMat1, class Triangle, class DiagonalStorage,
+         @\exposconcept{in-matrix}@ InMat2, @\exposconcept{out-matrix}@ OutMat>
+  void triangular_matrix_matrix_left_solve(ExecutionPolicy&& exec,
+                                           InMat1 A, Triangle t, DiagonalStorage d,
+                                           InMat2 B, OutMat X);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+triangular_matrix_matrix_left_solve(std::forward<ExecutionPolicy>(exec),
+                                    A, t, d, B, X, divides<void>{});
+\end{codeblock}
+\end{itemdescr}
+
+\begin{itemdecl}
+template<@\exposconcept{in-matrix}@ InMat1, class Triangle, class DiagonalStorage,
+         @\exposconcept{in-matrix}@ InMat2, @\exposconcept{out-matrix}@ OutMat, class BinaryDivideOp>
+  void triangular_matrix_matrix_right_solve(InMat1 A, Triangle t, DiagonalStorage d,
+                                            InMat2 B, OutMat X, BinaryDivideOp divide);
+template<class ExecutionPolicy,
+         @\exposconcept{in-matrix}@ InMat1, class Triangle, class DiagonalStorage,
+         @\exposconcept{in-matrix}@ InMat2, @\exposconcept{out-matrix}@ OutMat, class BinaryDivideOp>
+  void triangular_matrix_matrix_right_solve(ExecutionPolicy&& exec,
+                                            InMat1 A, Triangle t, DiagonalStorage d,
+                                            InMat2 B, OutMat X, BinaryDivideOp divide);
+\end{itemdecl}
+
+\begin{itemdescr}
+
+\pnum
+These functions perform multiple matrix solves,
+taking into account the \tcode{Triangle} and \tcode{DiagonalStorage} parameters
+that apply to the triangular matrix \tcode{A}\iref{linalg.general}.
+
+\pnum
+\mandates
+\begin{itemize}
+\item
+If \tcode{InMat1} has \tcode{layout_blas_packed} layout,
+then the layout's \tcode{Triangle} template argument has
+the same type as the function's \tcode{Triangle} template argument;
+\item
+\tcode{\exposid{possibly-multipliable}<OutMat, InMat1, InMat2>()} is \tcode{true}; and
+\item
+\tcode{\exposid{compatible-static-extents}<InMat1, InMat1>(0,1)} is \tcode{true}.
+\end{itemize}
+
+\pnum
+\expects
+\begin{itemize}
+\item
+\tcode{\exposid{multipliable}(X,A,B)} is \tcode{true}, and
+\item
+\tcode{A.extent(0) == A.extent(1)} is \tcode{true}.
+\end{itemize}
+
+\pnum
+\effects
+Computes $X'$ such that $X'A = B$,
+and assigns each element of $X'$ to the corresponding element of $X$.
+If no such $X'$ exists,
+then the elements of \tcode{X} are valid but unspecified.
+
+\pnum
+\complexity
+$O($ \tcode{B.extent(0)} $\cdot$ \tcode{B.extent(1)} $\cdot$ \tcode{A.extent(1)} $)$
+\begin{note}
+Since the triangular matrix is on the right,
+the desired \tcode{divide} implementation
+in the case of noncommutative multiplication
+is mathematically equivalent to $x y^{-1}$,
```

Comment:
you have a space here in the Latex source (between x and y), yet the rendering looks good (i.e. no space in pdf, so multiplication assumed) - this is better than having a space in the pdf (the space makes it look like 2 separate expressions).

... there still needs to be an association between /x/ and \tcode{x} tho

Reply:
Is this comment trying to express more than your general discomfort with implicitly equating math variables with C++ function parameters via [linalg.general] p1?  If so, I don't get it; please try to rephrase.

Side note: LaTeX math mode has special rendering rules, and spaces in the source don't necessarily turn into spaces in the resulting formula.

Reply:
Yes, but maybe moot now? The spaces in the Latex source were sometimes rendered in the pdf, making some mathy expressions look like two separate expressions (e.g. $A' = A + x x^T$ became "A' = A + x x^T" with a space between the two x's). But I can't find where those cases are in the latest pdf, so you fixed them already. I looked through all the "Effects: Computes" and they look fine now.

Reply:
Interesting, I'm not aware of any changes in that area, or even of the possibility of having an influence (absent `\ ` or similarly gross things).